src/common/ceph_json.cc

```diff
@@ -136,49 +121,53 @@ bool JSONObj::get_data(const string& key, data_val *dest)
 }
 
 /* accepts a JSON Array or JSON Object contained in
- * a JSON Spirit Value, v,  and creates a JSONObj for each
+ * a Boost.JSON value, v, and creates a Ceph JSONObj for each
  * child contained in v
  */
-void JSONObj::handle_value(Value v)
-{
-  if (v.type() == obj_type) {
-    Object temp_obj = v.get_obj();
-    for (Object::size_type i = 0; i < temp_obj.size(); i++) {
-      Pair temp_pair = temp_obj[i];
-      string temp_name = temp_pair.name_;
-      Value temp_value = temp_pair.value_;
-      JSONObj *child = new JSONObj;
-      child->init(this, temp_value, temp_name);
-      add_child(temp_name, child);
-    }
-  } else if (v.type() == array_type) {
-    Array temp_array = v.get_array();
-    Value value;
+void JSONObj::handle_value(boost::json::value v)
+{
+  if (auto op = v.if_object()) {
 
-    for (unsigned j = 0; j < temp_array.size(); j++) {
-      Value cur = temp_array[j];
-      string temp_name;
+	for (const auto& kvp : *op)
```

Comment:
Nope, let's honor the existing indentation. Please see https://github.com/ceph/ceph/blob/main/CodingStyle.

Reply:
I've tried to fix this at least somewhat, hopefully it will be a little be easier on your eyes. If we are truly concerned, maybe some light auto-formatting is an interesting idea.